cloud/db/services/organization-memberships.ts

```diff
@@ -0,0 +1,527 @@
+        organizationId,
+        targetUserId,
+      });
+    });
+  }
+
+  /**
+   * Updates a membership's role.
+   *
+   * Requires OWNER or ADMIN role. Cannot be used to:
+   * - Change a member's role to OWNER
+   * - Change an OWNER's role (owner is immutable)
+   * - Change your own role (no self-modification)
+   *
+   * Security: Path params (`organizationId`, `targetUserId`) are authoritative.
+   * The `data` type excludes these fields to prevent injection attacks.
+   *
+   * @param args.userId - The authenticated user performing the action
+   * @param args.organizationId - The organization (path param)
+   * @param args.targetUserId - The user whose membership to update (path param)
+   * @param args.data - Fields to update (only `role` allowed)
+   * @returns The updated membership
+   * @throws PermissionDeniedError - If violating owner/self constraints
+   * @throws NotFoundError - If the membership doesn't exist
+   * @throws DatabaseError - If the database operation fails
+   */
+  update({
+    userId,
+    organizationId,
+    targetUserId,
+    data,
+  }: {
+    userId: string;
+    organizationId: string;
+    targetUserId: string;
+    data: Pick<NewOrganizationMembership, "role">;
+  }): Effect.Effect<
+    PublicOrganizationMembership,
+    NotFoundError | PermissionDeniedError | DatabaseError
+  > {
+    return Effect.gen(this, function* () {
+      yield* this.authorize({ userId, action: "update", organizationId });
+
+      if (data.role === "OWNER") {
+        return yield* Effect.fail(
+          new PermissionDeniedError({
+            message: "Cannot change a member's role to owner",
+            resource: "organization membership",
+          }),
+        );
+      }
+
+      if (targetUserId === userId) {
+        return yield* Effect.fail(
+          new PermissionDeniedError({
+            message: "Cannot modify your own membership",
+            resource: "organization membership",
+          }),
+        );
+      }
+
+      const targetMembership = yield* this.getMembership({
+        userId: targetUserId,
+        organizationId,
+      });
+
+      if (!targetMembership) {
+        return yield* Effect.fail(
+          new NotFoundError({
+            message: "User is not a member of this organization",
+            resource: "organization membership",
+          }),
+        );
+      }
+
+      if (targetMembership.role === "OWNER") {
+        return yield* Effect.fail(
+          new PermissionDeniedError({
+            message: "Cannot change the role of an owner",
+            resource: "organization membership",
+          }),
+        );
+      }
+
+      return yield* this.baseService.update({
+        organizationId,
+        targetUserId,
+        data: { role: data.role },
+      });
+    });
+  }
+
+  /**
+   * Removes a member from an organization.
+   *
+   * Requires OWNER role. Cannot be used to:
+   * - Remove an OWNER (owner is immutable)
+   * - Remove yourself (no self-removal)
```

Comment:
I think it should be possible for a user to leave an organization (except for owner role, where they would need to transfer ownership or delete the org). Worth adding now?